torch/jit/__init__.py

```diff
@@ -805,6 +817,102 @@ def init_then_register(self, *args, **kwargs):
 
 if _enabled:
     class ScriptModule(with_metaclass(ScriptMeta, torch._C.ScriptModule, Module)):
+        r"""
+        The core data structure in Torch Script is the ``ScriptModule``. It is an analogue
+        of torch's nn.Module and represents an entire model as a tree of submodules.
+        Like normal modules, each individual module in a ScriptModule can have submodules,
+        parameters, and methods. In nn.Modules methods are implemented as Python functions,
+        but in ScriptModules methods are actually *Torch Script* functions, a statically-typed subset
+        of Python that contains all of PyTorch's built-in Tensor operations. This difference
+        allows your ScriptModules code to run without the need for a Python interpreter.
+
+        ScriptModules and the Torch Script functions inside of them can be created in
+        two ways:
+
+        tracing
+            Using ``torch.jit.trace``, you can take an existing module or python
+            function, provide example inputs, and we run the function, recording the
+            operations performed on all the tensors. We turn the resulting recording
+            into a Torch Script method that is installed as the ``forward`` method of a
+            ScriptModule. This module also contains any parameters that the original
+            module had as well.
+
+            Example::
+
+                import torch
+                def foo(x, y):
+                    return 2*x + y
+                traced_foo = torch.jit.trace(foo, (torch.rand(3), torch.rand(3)))
+
+            Example::
+
+                import torch
+                import torchvision
+                traced_net = torch.jit.trace(torchvision.models.resnet18(),
+                                             torch.rand(1, 3, 224, 224))
+
+        scripting
```

Comment:
Does this actually render nicely? You can add underlines to words to make them headers.